### ISSUE TITLE
Improve treatment of high-radius particles

### ISSUE THREAD

**Issue by RemiLehe** [fbpic/fbpic#369]
Currently, the treatment of high-radius particles (i.e. particles that are outside of the box radially) in FBPIC is not rigorous:
- High-radius particles are kept in the simulation (even though they are not inside the simulation domain anymore)
- They still gather the field, as if they were in the last radial cell.
- They still deposit their charge and current as if they were in the last radial cell ; charge deposited beyond `rmax` (because of the width of the particle shape) is wrapped back into the domain. While wrapping the charge makes sense close to the axis, it does not make much sense at high radius.

It would be good to correct this behavior.

**Comment by danielseipt** [comment]
I have also seen that these high-radius particles can re-enter the simulation box and, for instance, be trapped in a 2nd or 3rd wakefield bucket

**Comment by danielseipt** [comment]
Here is an example input deck that shows the returning particles. In the plot the grey dashed lines mark the boundary of the simulation box.

[bugreport_out_of_box_particles.log](https://github.com/fbpic/fbpic/files/3790051/bugreport_out_of_box_particles.log)


![bugreport_fast_00023750](https://user-images.githubusercontent.com/39627867/67877472-f9784200-fb0f-11e9-8b43-bec342072a85.png)


[bugreport_movie(5).avi.zip](https://github.com/fbpic/fbpic/files/3790151/bugreport_movie.5.avi.zip)

**Comment by RemiLehe** [comment]
Thanks a lot for sending this reproducer!
This should be relatively easy to fix (we should basically prevent the high-radius particles from gathering the fields). I'll try to fix this in the next few days.

**Comment by RemiLehe** [comment]
@danielseipt The issue that you are reporting should now be fixed in the `dev` branch of fbpic.

This fix will be part of the next release. But in the meantime, if you want to use it already, you can install the `dev` branch by doing:
```
pip uninstall fbpic
pip install git+https://github.com/fbpic/fbpic.git@dev
```

**Comment by MKirchen** [comment]
Closed by #398
